lib/rucio/core/replica.py

```diff
@@ -1462,63 +1463,108 @@ def delete_replicas(rse_id, files, ignore_availability=True, session=None):
-                                                                                                                                           models.DataIdentifier.availability == DIDAvailability.LOST)),
-                                               ~exists(select([1]).prefix_with("/*+ INDEX(REPLICAS REPLICAS_PK) */", dialect='oracle')).where(and_(models.RSEFileAssociation.scope == file['scope'], models.RSEFileAssociation.name == file['name']))))
-
-        src_condition.append(and_(models.Source.scope == file['scope'],
-                                  models.Source.name == file['name'],
-                                  models.Source.rse_id == rse_id))
+        replica_condition.append(
+            and_(models.RSEFileAssociation.scope == file['scope'],
+                 models.RSEFileAssociation.name == file['name']))
+
+        src_condition.append(
+            and_(models.Source.scope == file['scope'],
+                 models.Source.name == file['name'],
+                 models.Source.rse_id == rse_id))
 
     delta, bytes, rowcount = 0, 0, 0
 
     # WARNING : This should not be necessary since that would mean the replica is used as a source.
     for chunk in chunks(src_condition, 10):
-        rowcount = session.query(models.Source).\
-            filter(or_(*chunk)).\
+        rowcount = session.query(models.Source). \
+            filter(or_(*chunk)). \
             delete(synchronize_session=False)
 
     rowcount = 0
     for chunk in chunks(replica_condition, 10):
-        for (scope, name, rid, replica_bytes) in session.query(models.RSEFileAssociation.scope, models.RSEFileAssociation.name, models.RSEFileAssociation.rse_id, models.RSEFileAssociation.bytes).\
+        for (scope, name, rid, replica_bytes) in session.query(models.RSEFileAssociation.scope, models.RSEFileAssociation.name, models.RSEFileAssociation.rse_id, models.RSEFileAssociation.bytes). \
                 with_hint(models.RSEFileAssociation, "INDEX(REPLICAS REPLICAS_PK)", 'oracle').filter(models.RSEFileAssociation.rse_id == rse_id).filter(or_(*chunk)):
             bytes += replica_bytes
             delta += 1
 
-        rowcount += session.query(models.RSEFileAssociation).filter(models.RSEFileAssociation.rse_id == rse_id).filter(or_(*chunk)).delete(synchronize_session=False)
+        rowcount += session.query(models.RSEFileAssociation). \
+            filter(models.RSEFileAssociation.rse_id == rse_id). \
+            filter(or_(*chunk)). \
+            delete(synchronize_session=False)
 
     if rowcount != len(files):
         raise exception.ReplicaNotFound("One or several replicas don't exist.")
 
+    __update_metadata_after_replica_deletion(rse_id=rse_id, files=files, session=session)
+
+    # Decrease RSE counter
+    decrease(rse_id=rse_id, files=delta, bytes=bytes, session=session)
+
+
+@transactional_session
+def __update_metadata_after_replica_deletion(rse_id, files, session=None):
```

Comment:
I would call this `__cleanup_after_replica_deletion`
`metadata` has a bit of a double-loaded meaning in Rucio. I think cleanup is cleaner 😄

Reply:
Done